pkg/reconciler/revision/cruds_test.go

```diff
@@ -0,0 +1,381 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package revision
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	corev1informer "k8s.io/client-go/informers/core/v1"
+	kubefake "k8s.io/client-go/kubernetes/fake"
+	"knative.dev/pkg/ptr"
+	"knative.dev/serving/pkg/apis/serving"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+)
+
+var (
+	containerName = "my-container-name"
+
+	defaultRevision = &v1alpha1.Revision{
```

Comment:
You only pass this revision into `verifySecrets`. I think this can be greatly stripped down to just the minimum necessary to achieve that. TimeoutSeconds, readiness probes etc etc. are not needed to fullfill that, not even ObjectMeta is I think?

With that in mind I feel like inlining the minimal revision bits into the tests themselves without any factory functions might be the cleanest?